### DISCUSSION TITLE
Prometheus metrics on KEDA Metric Server are deprecated in favor of Prometheus metrics on KEDA Operator

### DISCUSSION THREAD

**Discussion by tomkerkhove** [kedacore/keda#3973] [Deprecations]
Prometheus metrics on KEDA Metric Server are deprecated in favor of Prometheus metrics on KEDA Operator.

This change was done to bring the metrics closer to the operator which is now integrating with 3rd party for improved performance and integration

## Impact
- End-users have to scrape metrics on KEDA operator
- Metric names have been renamed to use `keda_scaler_` prefix instead of `keda_metrics_adapter_`

Change was introduced in https://github.com/kedacore/keda/issues/3919.

## Timeline

- Dec 7th, 2022 - Deprecation
- July, 2023 - Removal in KEDA v2.11

**Docs:** https://keda.sh/docs/latest/operate/prometheus/#keda-operator
**Deprecation Issue:** https://github.com/kedacore/keda/issues/3972

**Comment by EppO** [comment]
> ## Timeline
> * Dec 7th, 2022 - Deprecation
> * **July, 2022** - Removal in KEDA v2.11
> 

I guess you meant July, 2023

**Reply by zroubalik** [reply]
Fixed, thanks!

**Comment by BojanZelic** [comment]
I recently filed an issue that's related to this - https://github.com/kedacore/keda/issues/4460 and was going to look into opening a PR for it, but I want to be aligned here before I do any implementation;

My idea is to capture generic request information and export it as a prometheus metric; Things like response code, duration, # of requests, ect... If the plan is to deprecate all prometheus metrics from the KEDA metrics server that means there will be no ability to export these types of metrics and it wouldn't make sense to do it from the operator;

**Reply by BojanZelic-DD** [reply]
nevermind on above, the depricated metrics are on port `9022`

**Reply by tomkerkhove** [reply]
I was just going to reply, but neverminding :)

**Reply by JorTurFer** [reply]
These metrics have been removed
